src/main/scala/org/apache/sysml/compiler/lang/source/DML.scala

```diff
@@ -0,0 +1,473 @@
+                else {
+                  method.name.decodedName.toTermName match {
+                    case u.TermName("to")    => s"${tgt(env)} + 1:${arg(env)} + 1"
+                    case u.TermName("until") => s"${tgt(env)} + 1:${arg(env)}"
+                    case u.TermName("%") => s"($module %% ${args(0)})" // modulo in dml is %%
+                    case u.TermName("&&") => s"($module & ${args(0)})" // && in dml is &
+                    case u.TermName("||") => s"($module | ${args(0)})" // || in dml is |
+                    case _ => s"($module ${method.name.decodedName} ${args(0)})"
+                  }
+                }
+              }
+
+              // matches apply methods with multiple arguments
+              case (Some(tgt), (x :: xs) :: Nil) if isApply(method) => {
+                val module = tgt(env)
+                val argString = args.mkString(" ")
+
+                if (module == "Array") {
+                  // sequence/array constructors
+                  s""""$argString""""
+                }
+
+                else if (module == "Matrix") {
+                  printConstructor(method, argss, env, false)
+                }
+
+                else if (module == "Vector") {
+                  printConstructor(method, argss, env, true)
+                }
+
+                else if (module == "package") {
+                  // builtins
+                  "builtin"
+                }
+
+                  // apply as indexing - convert Scala 0-based to DML 1-based
+                else {
+                  val rows = args(0) // rows
+                  val cols = args(1) // columns
+
+                  (rows, cols) match {
+                    case (":::", c) if c.contains(":")  => s"$module[,$c]"
+                    case (":::", c)                     => s"$module[,$c + 1]"
+                    case (r, ":::") if r.contains(":")  => s"$module[$r,]"
+                    case (r, ":::")                     => s"$module[$r + 1,]"
+                    case (r, c) if  r.contains(":") &&
+                                    c.contains(":")     => s"$module[$r,$c]"
+                    case (r, c) if r.contains(":")      => s"$module[$r, $c + 1]"
+                    case (r, c) if c.contains(":")      => s"$module[$r + 1, $c]"
+                    case (r, c)                         => s"as.scalar($module[$r + 1,$c + 1])"
+                  }
+                }
+              }
+
+              // matches apply methods with multiple arguments
+              case (Some(tgt), (x :: xs) :: Nil) if isUpdate(method) => {
+                val module = tgt(env)
+
+                // update on matrix objects (left indexing): A[r, c] = v === A.update(r, c, v)
+                val rows = args(0) // rows
+                val cols = args(1) // columns
+                val value = args(2) // value to update with
+
+                (rows, cols) match {
+                  case (":::", c) if c.contains(":")  => s"$module[,$c] = $value"
```

Comment:
Duplicate of logic in the case statement about 10 lines earlier. Consider breaking this logic out into a separate method.